src/Sylius/Bundle/CoreBundle/CatalogPromotion/Checker/InForTaxonsScopeVariantChecker.php

```diff
@@ -59,7 +59,8 @@ private function getTaxonChildrenCodes(TaxonInterface $taxon): array
         $childrenCodes = [];
 
         foreach ($this->taxonTreeRepository->children($taxon) as $taxonChild) {
-            $childrenCodes[] = $taxonChild->getCode();
+            Assert::isInstanceOf($taxonChild, TaxonInterface::class);
+            $childrenCodes[] = (string) $taxonChild->getCode();
```

Comment:
Assert `notNull($taxonChild->getCode())` is better then pottensionaly type cast null to string.